kernel/src/syscall/sethostname.rs

```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use super::SyscallReturn;
+use crate::{net::HOSTNAME, prelude::*};
+
+pub fn sys_sethostname(name: Vaddr, len: usize, ctx: &Context) -> Result<SyscallReturn> {
+    let mut buf = vec![0u8; len];
```

Comment:
To make the system call more robust, check `len` against an upper bound.

Reply:
Good catch! I've added a `HOST_NAME_MAX` constant set to 64 to match Linux